vllm/engine/llm_engine.py

```diff
@@ -116,12 +115,13 @@ def __init__(
         # Create the scheduler.
         self.scheduler = Scheduler(scheduler_config, cache_config)
 
-        # Logging.
-        self.last_logging_time = 0.0
-        # List of (timestamp, num_tokens)
-        self.num_prompt_tokens: List[Tuple[float, int]] = []
-        # List of (timestamp, num_tokens)
-        self.num_generation_tokens: List[Tuple[float, int]] = []
+        # Metric Logging.
+        if self.log_stats:
+            self.metrics_logger = MetricsLogger(
+                metrics=METRICS_REGISTRY,
+                local_logger=logger,
```

Comment:
Usually the logger is not passed like argument. in The MetricsLogger you can just initiate the logger again.

Reply:
Happy to make this change.